### PR TITLE
Add handling for edge cases in nuimo_app for Sonos

### PR DESCRIPTION
**Pull request by neelsenic**

### CODE REVIEW

File: senic_hub/nuimo_app/components/sonos.py [hunk 1]
```diff
@@ -80,29 +82,40 @@ def update_state(self):
...
         elif self.state in [self.STATE_PAUSED, self.STATE_STOPPED]:
             self.play()
+            logger.debug("Play started/resumed by self.pause() on button press.")
+
+        elif self.state == None:
+            self.nuimo.display_matrix(matrices.ERROR)
+            logger.debug("No Active connection with Host.")
 
-        logger.debug("state toggle: %s", self.state)
+        logger.debug("state toggle : %s", self.state)
```
Comment (rudolfs) [inline senic_hub/nuimo_app/components/sonos.py]: Why the extra space?

File: senic_hub/nuimo_app/components/sonos.py [hunk 2]
```diff
@@ -80,29 +82,40 @@ def update_state(self):
...
+                self.nuimo.display_matrix(matrix, fading=True, ignore_duplicates=True)
 
-        except SoCoException:
-            self.nuimo.display_matrix(matrices.ERROR)
+            except SoCoException:
+                self.nuimo.display_matrix(matrices.ERROR)
 
-        self.last_request_time = time()
+            self.last_request_time = time()
+
```
Comment (rudolfs) [inline senic_hub/nuimo_app/components/sonos.py]: No newlines needed between conditional branches.

File: senic_hub/nuimo_app/components/sonos.py [hunk 3]
```diff
@@ -80,29 +82,40 @@ def update_state(self):
...
             self.pause()
+            logger.debug("Play Paused by self.pause() on button press.")
 
         elif self.state in [self.STATE_PAUSED, self.STATE_STOPPED]:
             self.play()
+            logger.debug("Play started/resumed by self.pause() on button press.")
+
+        elif self.state == None:
+            self.nuimo.display_matrix(matrices.ERROR)
+            logger.debug("No Active connection with Host.")
```
Comment (rudolfs) [inline senic_hub/nuimo_app/components/sonos.py]: Let's keep conditionals compact, like in the rest of the codebase:
```
if
elif
elif
...
```

File: senic_hub/nuimo_app/components/sonos.py (hunk 2 above)
Comment (neelsenic) [inline senic_hub/nuimo_app/components/sonos.py]: 6a168084cd02141d8331e1759980f60bd14db254

**Pull request merged**